### PR TITLE
Certik RBS-6/7 Replace the sponsorship check with an additional whitelist

### PR DESCRIPTION
**Pull request by bbenligiray**
I didn't implement template-specific whitelisting. If that is needed, it can be implemented in a proxy contract and the sponsor can permit that to request updates instead.

Tests are not implemented/fixed

### CODE REVIEW

**Comment by Ashar2shahid** [comment]
I'm working on the tests for this, I'll push it on this branch because the implementation is more or less the same

**Comment by acenolaza** [comment]
This change is pretty straightforward if I'm not mistaken. We just need to replicate sponsorship flow from AirnodeRrp in here.

File: packages/protocol/contracts/rrp/requesters/interfaces/IRrpBeaconServer.sol [hunk 1]
```diff
@@ -42,4 +53,9 @@ interface IRrpBeaconServer is IMetaAdminnable, IWhitelister {
         external
         view
         returns (int224 value, uint32 timestamp);
+
+    function sponsorToUpdateRequesterToPermissonStatus(
```
Comment (Ashar2shahid) [inline packages/protocol/contracts/rrp/requesters/interfaces/IRrpBeaconServer.sol:57]: Just curious but is it standard practice to explicitly state public mappings/structs in the interface? Shouldn't you be able to query them regardless? I've seen this done in the TemplateUtils as well.

File: packages/protocol/contracts/rrp/requesters/RrpBeaconServer.sol [hunk 2]
```diff
@@ -42,12 +46,35 @@ contract RrpBeaconServer is
...
+    /// @notice Called by the sponsor to set the update request permission
+    /// status of an account
+    /// @param updateRequester Update requester address
+    /// @param updatePermissionStatus Update permission status of the update
+    /// requester
+    function setUpdatePermissionStatus(
+        address updateRequester,
+        bool updatePermissionStatus
+    ) external override {
+        require(updateRequester != address(0), "updateRequester address zero");
```
Comment (Ashar2shahid) [inline packages/protocol/contracts/rrp/requesters/RrpBeaconServer.sol:58]: I have added this check just to be sure.

**Review by Ashar2shahid — APPROVED** [review]
I've updated the tests and added an extra check. Can have a look

**Comment by bbenligiray** [comment]
Agreed and closed.

**Pull request closed without merging**